modules/apps/29-fee/ibc_middleware.go

```diff
@@ -333,6 +333,61 @@ func (im IBCMiddleware) OnTimeoutPacket(
 	return im.app.OnTimeoutPacket(ctx, packet, relayer)
 }
 
+// OnChanUpgradeInit implements the IBCMiddleware interface
+func (im IBCMiddleware) OnChanUpgradeInit(
+	ctx sdk.Context,
+	order channeltypes.Order,
+	connectionHops []string,
+	portID, channelID string,
+	sequence uint64,
+	counterparty channeltypes.Counterparty,
+	version, previousVersion string,
+) (string, error) {
+	return im.app.OnChanUpgradeInit(ctx, order, connectionHops, portID, channelID, sequence, counterparty, version, previousVersion)
```

Comment:
Note: we need something more complex here later

Reply:
Yep, see here for a WIP branch I have started with some tests. 
I think we need the handshake implementation before moving forward with the ics29 callbacks and tests (in order to get them passing).